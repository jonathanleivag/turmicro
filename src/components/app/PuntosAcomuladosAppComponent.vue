<script setup lang="ts">
import { defineProps, onMounted } from 'vue'
import Web3 from 'web3/dist/web3.min.js'

import { Turmicro } from '../../interfaces/ITurmicro'
import { useStore } from '../../store'
import CardUiComponent from '../ui/CardUiComponent.vue'

interface IProps {
  account: string
  deployContract: Turmicro
  web3: Web3
}

const props = defineProps<IProps>()
const store = useStore()

onMounted(() => {
  getPuntosEnEther()
})

const getPuntosEnEther = async () => {
  store.dispatch('puntosAcomulados/setPuntosAcomulados', {
    deployContract: props.deployContract,
    account: props.account,
    web3: props.web3
  })
}

const canjearPuntos = async () => {
  await props.deployContract.transferirPuntosEnEther({ from: props.account })
  getPuntosEnEther()
}
</script>

<template>
  <CardUiComponent title="Puntos acomulados">
    <div>
      <p>
        <span class="negrita">Puntos</span>:
        {{ store.state.puntosAcomulados.puntos }}
      </p>
      <p>
        <span class="negrita">Valor Punto</span>:
        {{ store.state.puntosAcomulados.valorPunto }} ETH
      </p>
      <p>
        <span class="negrita">Acomulado</span>:
        {{ store.state.puntosAcomulados.puntosAcomulados }} ETH
      </p>
    </div>
    <div class="card__mantenedor_boton">
      <button
        @click="canjearPuntos"
        typer="button"
        class="card__comprar card__comprar--puntos"
      >
        Canjear Puntos
      </button>
    </div>
  </CardUiComponent>
</template>
