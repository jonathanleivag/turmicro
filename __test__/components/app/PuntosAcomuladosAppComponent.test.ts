import { mount } from '@vue/test-utils'

import PuntosAcomuladosAppComponent from '../../../src/components/app/PuntosAcomuladosAppComponent.vue'

describe('PuntosAcomuladosAppComponent', () => {
  const wrapper = mount(PuntosAcomuladosAppComponent)

  test('Snapshot de PuntosAcomuladosAppComponent', () => {
    expect(PuntosAcomuladosAppComponent).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('El titulo tiene que ser Puntos acomulados', () => {
    expect(wrapper.find('h3').text()).toBe('Puntos acomulados')
  })
})
