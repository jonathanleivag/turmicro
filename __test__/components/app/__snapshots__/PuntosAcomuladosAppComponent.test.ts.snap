// Vitest Snapshot v1

exports[`BalanceAppComponent > Snapshot de PuntosAcomuladosAppComponent 1`] = `
"<div class=\\"card\\">
  <div class=\\"card__header\\">
    <h3 class=\\"card__title\\">Puntos acomulados</h3>
  </div>
  <div class=\\"card__body\\"></div>
</div>"
`;

exports[`PuntosAcomuladosAppComponent > Snapshot de PuntosAcomuladosAppComponent 1`] = `
"<div class=\\"card\\">
  <div class=\\"card__header\\">
    <h3 class=\\"card__title\\">Puntos acomulados</h3>
  </div>
  <div class=\\"card__body\\"></div>
</div>"
`;
